pkg/virt-controller/watch/vmi.go

```diff
@@ -557,22 +582,29 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		return &syncErrorImpl{err, FailedPVCVolumeSourceMisusedReason}
 	}
 
+	dataVolumesReady, isWaitForFirstConsumer, syncErr := c.handleSyncDataVolumes(vmi, dataVolumes)
+	if syncErr != nil {
+		return syncErr
+	}
 	if !podExists(pod) {
 		// If we came ever that far to detect that we already created a pod, we don't create it again
 		if !vmi.IsUnprocessed() {
 			return nil
 		}
 
 		// ensure that all dataVolumes associated with the VMI are ready before creating the pod
-		dataVolumesReady, syncErr := c.handleSyncDataVolumes(vmi, dataVolumes)
-		if syncErr != nil {
-			return syncErr
-		} else if !dataVolumesReady {
+		if !dataVolumesReady {
 			log.Log.V(3).Object(vmi).Infof("Delaying pod creation while DataVolume populates")
 			return nil
 		}
-
-		templatePod, err := c.templateService.RenderLaunchManifest(vmi)
+		var templatePod *k8sv1.Pod
+		var err error
+		if isWaitForFirstConsumer {
+			log.Log.V(3).Object(vmi).Infof("Scheduling temporary pod for WaitForFirstConsumer DV")
+			templatePod, err = c.templateService.RenderLaunchManifestNoVm(vmi)
```

Comment:
I thought about this a bit over the weekend. One thing that this PR doesn't handle is a situation where the tempPod fails.  It's techinically possible for a pod to be scheduled to a node, and then fail to launch on that node for reasons unrelated to the workload. 

Because of this, a tempPod could be created and never technically serve it's purpose, then I believe the VMI will get stuck in pending forever.

To fix this edge case, we could make the tempPod a k8s Job instead, which guarantees that the pod will be retried until it eventually succeeds somewhere. You can look at `pkg/virt-operator/kubevirt.go` for an example of how we used a job in another controller.

Reply:
I am looking at this right now, in the meantime a rebased/fresh version was pushed to check if there are any failing tests due to new featureflag being set.

Reply:
good, so I can easily wrap a pod in a job, but as I did not use jobs yet I have a few questions
- do you think we need some special settings for backoff and restart 
- will the pod nicely be deleted after job finishes or do I need to find and kill a job or a pod later?

Reply:
ok, so after cleaning up other problems, I took a try at this, and this looks like a doable and not so hard task, but we for sure can not make it today/tomorrow, there are some rules/access rights to change, tests to update and small details to analyze. For example I get this when trying to create a job:
```
obs.batch is forbidden: User "system:serviceaccount:kubevirt:kubevirt-controller" cannot create resource "jobs" in API group "batch" in the namespace "kubevirt-test-default"
```

So @davidvossel, @aglitke, I am asking, what is the risk for leaving it as is right now as the `pod` (this new feature that fixes some other problems), and doing a conversion to `job` after merge as a new follow-up task.

Reply:
> So @davidvossel, @aglitke, I am asking, what is the risk for leaving it as is right now as the pod (this new feature that fixes some other problems), and doing a conversion to job after merge as a new follow-up task.

The alternative to the job logic is to tie the lifecycle of the VMI to the temp pod... so if the temp pod fails for any reason, immediately fail the VMI just like we would the real pod. 

The situation we don't want to be in is having a VMI stuck in pending indefinitely due to a failed temp pod that will never get retried and will never cause the VMI to move out of pending.